mvpoly/src/prime.rs

```diff
@@ -805,28 +804,13 @@ impl<F: PrimeField, const N: usize, const D: usize> From<F> for Dense<F, N, D> {
     }
 }
 
-impl<F: PrimeField, const N: usize, const D: usize>
-    From<ConstantExprInner<F, BerkeleyChallengeTerm>> for Dense<F, N, D>
+impl<F: PrimeField, const N: usize, const D: usize, ChallengeTerm>
+    From<ConstantExprInner<F, ChallengeTerm>> for Dense<F, N, D>
 {
-    fn from(expr: ConstantExprInner<F, BerkeleyChallengeTerm>) -> Self {
+    fn from(expr: ConstantExprInner<F, ChallengeTerm>) -> Self {
         match expr {
-            // The unimplemented methods might be implemented in the future if
-            // we move to the challenge into the type of the constant
-            // terms/expressions
-            // Unrolling for visibility
-            ConstantExprInner::Challenge(BerkeleyChallengeTerm::Alpha) => {
-                unimplemented!("The challenge alpha is not supposed to be used in this context")
-            }
-            ConstantExprInner::Challenge(BerkeleyChallengeTerm::Beta) => {
-                unimplemented!("The challenge beta is not supposed to be used in this context")
-            }
-            ConstantExprInner::Challenge(BerkeleyChallengeTerm::Gamma) => {
-                unimplemented!("The challenge gamma is not supposed to be used in this context")
-            }
-            ConstantExprInner::Challenge(BerkeleyChallengeTerm::JointCombiner) => {
-                unimplemented!(
-                    "The challenge joint combiner is not supposed to be used in this context"
-                )
+            ConstantExprInner::Challenge(_) => {
+                unimplemented!("Challenges are not supposed to be used in this context for now")
```

Comment:
Note that it will be soon. I haven't encountered the case I need it. I'm building step by step.